src/server/game/Entities/Object/Object.cpp

```diff
@@ -2697,6 +2697,13 @@ void WorldObject::GetCreatureListWithEntryInGrid(std::list<Creature*>& creatureL
     cell.Visit(pair, visitor, *(this->GetMap()), *this, maxSearchRange);
 }
 
+void WorldObject::GetPlayerListInGrid(std::list<Player*>& playerList, float maxSearchRange) const
+{
+    Trinity::AnyPlayerInObjectRangeCheck checker(this, maxSearchRange);
+    Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(this, playerList, checker);
+    this->VisitNearbyWorldObject(maxSearchRange, searcher);
```

Comment:
https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp#L523-L526
https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/scripts/Spells/spell_holiday.cpp#L80-L83
https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/scripts/Spells/spell_quest.cpp#L1537-L1540

I think it's worth to replace that code by this new method.